### PR TITLE
Remove long-deprecated LEGACY mode

### PR DESCRIPTION
**Pull request by agoode**
Removing this mode allows both `ENABLE_AVAHI` and `ENABLE_LEGACY` defines to be removed and eliminates a lot of code.

### CODE REVIEW

**Comment by agoode** [comment]
Can you take a look and tell me your thoughts? Not sure how you'd like to do code reviews in general, but I'd like to get your (or someone's) review on non-trivial changes. Thanks!

**Comment by lathiat** [comment]
Reviewing now

**Comment by agoode** [comment]
How's it going?

I can recruit some more reviewers to this project but can we move it to an organization first? (It simplifies the permissions.)

I propose "avahi-zeroconf" since "avahi" is taken. https://github.com/account/organizations/new

**Comment by agoode** [comment]
I would like to make some progress towards a new release. I will start pushing these cleanup and legacy removal patches to the master branch, and we can revert later if it looks like a bad idea.

File: src/Makefile.am [hunk 1]
```diff
@@ -51,18 +51,8 @@ lib_LTLIBRARIES += \
...
-if ENABLE_AVAHI
-sources += avahi.c avahi.h
-noinst_PROGRAMS += avahi-test
-endif
-
-if ENABLE_LEGACY
-sources += query.c query.h dns.c dns.h
-noinst_PROGRAMS += mdns-test
-endif
+noinst_PROGRAMS = nss-test avahi-test
```
Comment (vapier) [inline src/Makefile.am]: you want check_PROGRAMS.  noinst get built when you run `make`, but check get built only when you `make check`.

Comment (agoode) [inline src/Makefile.am]: These aren't proper tests, they have hardcoded addresses in them. They could be made into tests but currently will just fail after a while.

Comment (vapier) [inline src/Makefile.am]: i'm not sure that makes a difference to my proposal

Comment (agoode) [inline src/Makefile.am]: Ah, I thought `make check` would run them. Thanks for that.

**Comment by vapier** [comment]
you have to also list programs in `TESTS` to run them.  if people want these programs, they could also do `make nss-test` ... granted it's a bit more work.

File: src/Makefile.am (hunk 1 above)
Comment (vapier) [inline src/Makefile.am]: you have to also list programs in `TESTS` to run them.  if people want these programs, they could also do `make nss-test` ... granted it's a bit more work.

**Comment by agoode** [comment]
That's fine, we don't actually want to run them as they require remote hosts to be set up in a certain way. But they are useful to build when needed.

**Comment by agoode** [comment]
When you're happy with the code, hit the "Approve" option in Review changes and it will tag the request as reviewed.

**Pull request merged**